src/provider/providerWrapper.ts

```diff
@@ -197,29 +207,36 @@ export class ProviderWrapper {
     }
 
     switch (erc725Version) {
-      case ERC725_VERSION.ERC725:
-        return this._getAllData(address, keyHashes);
+      case ERC725_VERSION.ERC725_v5:
+        return this._getAllDataGeneric(
+          address,
+          keyHashes,
+          Method.GET_DATA_BATCH,
+        );
+      case ERC725_VERSION.ERC725_v2:
+        return this._getAllDataGeneric(address, keyHashes, Method.GET_DATA);
       case ERC725_VERSION.ERC725_LEGACY:
         return this._getAllDataLegacy(address, keyHashes);
       default:
         return [];
     }
   }
 
-  private async _getAllData(
+  private async _getAllDataGeneric(
     address: string,
     keyHashes: string[],
+    method: Method,
```

Comment:
Very cool! You could even be more precise here, as we really need only 2 methods.